deepcell_spots/applications/polaris.py

```diff
@@ -23,227 +23,142 @@
-
-        return image
-
-    def _predict_segmentation(self,
-                              image,
-                              batch_size=4,
-                              image_mpp=None,
-                              pad_mode='constant',
-                              preprocess_kwargs={},
-                              postprocess_kwargs={}):
-        """Generates a list of coordinate spot locations of the input running
-        prediction with appropriate pre and post processing functions.
-        This differs from parent Application class which returns a labeled image.
-        Input images are required to have 4 dimensions
-        ``[batch, x, y, channel]``. Additional empty dimensions can be added
-        using ``np.expand_dims``.
-        Args:
-            image (numpy.array): Input image with shape
-                ``[batch, x, y, channel]``.
-            batch_size (int): Number of images to predict on per batch.
-            image_mpp (float): Microns per pixel for ``image``.
-            pad_mode (str): The padding mode, one of "constant" or "reflect".
-            preprocess_kwargs (dict): Keyword arguments to pass to the
-                pre-processing function.
-            postprocess_kwargs (dict): Keyword arguments to pass to the
-                post-processing function.
-        Raises:
-            ValueError: Input data must match required rank, calculated as one
-                dimension more (batch dimension) than expected by the model.
-            ValueError: Input data must match required number of channels.
-        Returns:
-            numpy.array: Coordinate spot locations
-        """
-        # Check input size of image
-        if len(image.shape) != self.required_rank:
-            raise ValueError('Input data must have {} dimensions. '
-                             'Input data only has {} dimensions'.format(
-                                 self.required_rank, len(image.shape)))
-
-        if image.shape[-1] != self.required_channels:
-            raise ValueError('Input data must have {} channels. '
-                             'Input data only has {} channels'.format(
-                                 self.required_channels, image.shape[-1]))
-
-        # Resize image, returns unmodified if appropriate
-        resized_image = self._resize_input(image, image_mpp)
+    def __init__(self,
+                 segmentation_model=None,
+                 segmentation_compartment='cytoplasm',
+                 spots_model=None):
 
-        # Generate model outputs
-        output_images = self._run_model(
-            image=resized_image, batch_size=batch_size,
-            pad_mode=pad_mode, preprocess_kwargs=preprocess_kwargs
-        )
+        self.spots_app = SpotDetection(model=spots_model)
 
-        # Resize output_images back to original resolution if necessary
-        label_image = self._resize_output(output_images, image.shape)
+        valid_compartments = ['cytoplasm', 'nucleus', 'None']
```

Comment:
I don't like that `None` is specified as a string instead of the `None` python object. Maybe reconsider how this is being used or check in other apps to see what the convention has been.